drv/lpc55-spi-server/src/main.rs

```diff
@@ -63,74 +369,339 @@ fn main() -> ! {
     );
 
     spi.enable();
-
-    let mut a_bytes: [u8; 8] = [0xaa; 8];
-    let mut b_bytes: [u8; 8] = [0; 8];
-
     sys_irq_control(1, true);
 
-    let mut tx = &mut a_bytes;
-    let mut rx = &mut b_bytes;
-
-    let mut tx_cnt = 0;
-    let mut rx_cnt = 0;
+    // Configure ROT_IRQ
+    let rot_irq = Pin::PIO0_18; // XXX Should be in app.toml
+                                // Direction must be set after other pin configuration.
+    gpio.set_dir(rot_irq, Direction::Output).unwrap_lite();
+    // At the begining, we are ready to receive and have nothing to transmit.
+    // Ensure that ROT_IRQ is not asserted
+    gpio.set_val(rot_irq, Value::One).unwrap_lite();
+    ringbuf_entry!(Trace::DeassertRotIrq);
+
+    // Configure SP_RESET
+    // Reset is an input to detect SP internal reset but can be an driven low
+    // to reset the SP.
+
+    let sp_reset = Pin::PIO0_9; // XXX Should be in app.toml
+    gpio.set_dir(sp_reset, Direction::Input).unwrap_lite();
+    // TODO: SP reset and reset detection don't have to be here,
+    // However, SP reset will affect Sprockets state, so here may be good.
+    // TODO: Detect SP reset events and invalidate any trust/session that may exist.
+    // TODO: Drive SP reset when appropriate.
+
+    let mut tx = [0u8; SPI_RSP_BUF_SIZE];
+    let mut tctx = &mut TxContext::new(&mut tx[..]);
+    let mut rx = [0u8; SPI_REQ_BUF_SIZE];
+    let mut rctx = &mut RxContext::new(&mut rx[..]);
 
-    spi.drain();
     spi.enable_rx();
-    spi.enable_tx();
-
-    let mut tx_done = false;
-    let mut rx_done = false;
+    spi.disable_tx(); // Ignore Tx interrupts for now.
+    spi.send_u8(0x00); // Throw away data, but this sets bit-level framing.
+    spi.drain(); // Nothing in the FIFOs is interesting yet.
```

Comment:
This seems odd – does this introduce a race between `0x00` being sent out by the SPI peripheral and the FIFO being cleared by `spi.drain()`?